custom/demographics_print.php

```diff
@@ -0,0 +1,593 @@
+require_once("$srcdir/options.inc.php");
+require_once("$srcdir/patient.inc");
+require_once("$srcdir/formdata.inc.php");
+require_once("$srcdir/formatting.inc.php");
+
+if (empty($encounter)) die("Please select or create a visit!");
+
+$PDF_OUTPUT   = true;
+
+$date_init = "";
+
+//THIS IS A VERSION OF GENERATE_PRINT_FIELD() THAT HAS BEEN REVERSED TO SHOW DROP-DOWN LISTS AS SINGLE FIELDS.
+//USED ONLY BY MSI WHEN PRODUCING ITS VERSION OF THE PRINT_DEMOGRAPHICS REPORT.
+//
+function generate_print_field_msi($frow, $currvalue) {
+    global $rootdir, $date_init, $PDF_OUTPUT;
+
+    $s = "";
+    $currescaped = htmlspecialchars($currvalue, ENT_QUOTES);
+
+    $data_type   = $frow['data_type'];
+    $field_id    = $frow['field_id'];
+    $list_id     = $frow['list_id'];
+    $fld_length  = $frow['fld_length'];
+
+    $description = htmlspecialchars(xl_layout_label($frow['description']), ENT_QUOTES);
+
+    // Can pass $frow['empty_title'] with this variable, otherwise
+    //  will default to 'Unassigned'.
+    // If it is 'SKIP' then an empty text title is completely skipped.
+    $showEmpty = true;
+    if (isset($frow['empty_title'])) {
+        if ($frow['empty_title'] == "SKIP") {
+            //do not display an 'empty' choice
+            $showEmpty = false;
+            $empty_title = "Unassigned";
+        }
+        else {     
+            $empty_title = $frow['empty_title'];
+        }
+    }
+
+    // generic single-selection list
+    //
+    if ($data_type == 1 || $data_type == 26) {
+        if (empty($fld_length)) {
+            if ($list_id == 'titles') {
+                $fld_length = 3;
+            } else {
+                $fld_length = 10;
+            }
+        }
+        $tmp = '';
+        if ($currvalue) {
+            $lrow = sqlQuery("SELECT title FROM list_options " .
+                "WHERE list_id = '$list_id' AND option_id = '$currvalue' AND activity = 1");
+            $tmp = xl_list_label($lrow['title']);
+            if (empty($tmp)) $tmp = "($currvalue)";
+        }
+        if ($tmp === '') $tmp = '&nbsp;';
+        $s .= $tmp;
+    }
+
+    // simple text field
+    if ($data_type == 2 || $data_type == 15) {
+        if ($currescaped === '') $currescaped = '&nbsp;';
+        $s .= $currescaped;
+    }
+
+    // long or multi-line text field
+    else if ($data_type == 3) {
+        $s .= "<textarea" .
+            " cols='$fld_length'" .
+            " rows='" . $frow['max_length'] . "'>" .
+            $currescaped . "</textarea>";
+    }
+
+    // date
+    else if ($data_type == 4) {
+        if ($currescaped === '') $currescaped = '&nbsp;';
+        $s .= oeFormatShortDate($currescaped);
+    }
+
+    // provider list
+    else if ($data_type == 10 || $data_type == 11) {
+        $tmp = '';
+        if ($currvalue) {
+            $urow = sqlQuery("SELECT fname, lname, specialty FROM users " .
+                "WHERE id = '$currvalue'");
+            $tmp = ucwords($urow['fname'] . " " . $urow['lname']);
```

Comment:
best to not cap stuff since breaks with UTF8 encoding
should escpae this with text()